### PR TITLE
CHI-469 - [Case View] - Loading indicator

### PR DESCRIPTION
**Pull request by andvirga**
Jira related: https://bugs.benetech.org/browse/CHI-469

Primary reviewer: @murilovmachado 

Changes description:
- Moved `useEffect` out from `Timeline.tsx` to `Case.tsx` (that component is the responsible of showing/hiding the circular layout spinner).
- Passing setLoaded as callback into `getActivities` API.
- Converted `Timeline.jsx` to Typescript and added some useful types (`Activity`, `NoteActivity`, etc.).

### CODE REVIEW

**Review by GPaoloni — APPROVED** [review]
Looks good, just one comment.

File: plugin-hrm-form/src/states/case/types.ts [hunk 1]
```diff
@@ -86,6 +85,34 @@ export type CaseActionType =
...
+  text: string;
+  referral: {
+    date: string;
+    comments: string;
+    referredTo: string;
+  };
+  twilioWorkedId: string;
+};
+
+export type ConnectedCaseActivity = {
```
Comment (murilovmachado) [inline plugin-hrm-form/src/states/case/types.ts:110]: Great!

File: plugin-hrm-form/src/components/case/Case.tsx [hunk 2]
```diff
@@ -82,9 +84,45 @@ const Case: React.FC<Props> = props => {
...
 
+  useEffect(() => {
+    /**
+     * Gets the activities timeline from current caseId
+     * If the case is just being created, adds the case's description as a new activity.
+     */
+    const getTimeline = async () => {
+      const { task, form, connectedCaseState } = props;
+
+      const activities = await getActivities(connectedCaseState.connectedCase.id, setLoading);
```
[the remark this answers is not in the document]
Comment (murilovmachado) [reply plugin-hrm-form/src/components/case/Case.tsx:99]: I agree with Gian here.

File: plugin-hrm-form/src/components/case/Case.tsx [hunk 3]
```diff
@@ -308,7 +346,7 @@ const Case: React.FC<Props> = props => {
               />
             </Box>
             <Box marginLeft="25px" marginTop="25px">
-              <Timeline caseObj={connectedCase} task={task} form={form} status={status} />
+              <Timeline timeline={timeline} caseObj={connectedCase} task={task} form={form} status={status} />
```
Comment (murilovmachado) [inline plugin-hrm-form/src/components/case/Case.tsx:349]: I'd rename `timeline` props to `activities` or `caseActivities` or `timelineActivities`.

**Review by murilovmachado — APPROVED** [review]
Great to see moving some files from `jsx` to `tsx`. Left a few comments, but it's excellent work. Good job.

File: plugin-hrm-form/src/components/case/Case.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (andvirga) [reply plugin-hrm-form/src/components/case/Case.tsx:99]: Done, I've moved out that callback, thanks for your feedback

**Pull request merged**